src/states/bootstrapping.rs

```diff
@@ -227,31 +225,30 @@ impl Bootstrapping {
     }
 
     fn handle_timeout(&mut self, token: u64) {
-        if let Some((bootstrap_id, bootstrap_token)) = self.bootstrap_connection {
+        if let Some((ref bootstrap_id, bootstrap_token)) = self.bootstrap_connection {
             if bootstrap_token == token {
                 debug!(
                     "{:?} Timeout when trying to bootstrap against {:?}.",
                     self, bootstrap_id
                 );
-
-                self.rebootstrap();
+            } else {
+                return;
             }
+        } else {
+            return;
         }
+        self.rebootstrap();
```

Comment:
It's because of borrow checker. When we take a reference to `bootstrap_id` which is part of `self.bootstrap_connection`, we borrow `self` and can't call `self.rebootstrap()` in that scope